pallets/pallet-lottery/src/lib.rs

```diff
@@ -0,0 +1,1077 @@
+
+    /// NOTE: the total pot is the total number of KMA eligible to win in the current drawing cycle
+    #[pallet::storage]
+    #[pallet::getter(fn total_pot)]
+    pub(super) type TotalPot<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_users)]
+    pub(super) type TotalUsers<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn is_rebalancing)]
+    pub(super) type RebalanceInProgress<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn active_balance_per_user)]
+    pub(super) type ActiveBalancePerUser<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, BalanceOf<T>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn unclaimed_winnings_by_account)]
+    pub(super) type UnclaimedWinningsByAccount<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, BalanceOf<T>, OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_unclaimed_winnings)]
+    pub(super) type TotalUnclaimedWinnings<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    #[derive(Clone, Encode, Decode, TypeInfo)]
+    pub(super) struct UnstakingCollator<AccountId, BlockNumber> {
+        pub account: AccountId,
+        pub since: BlockNumber,
+    }
+
+    #[pallet::storage]
+    pub(super) type UnstakingCollators<T: Config> =
+        StorageValue<_, Vec<UnstakingCollator<T::AccountId, T::BlockNumber>>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn remaining_unstaking_balance)]
+    pub(super) type RemainingUnstakingBalance<T: Config> =
+        StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    #[derive(Clone, Debug, Eq, PartialEq, Encode, Decode, TypeInfo)]
+    pub struct Request<AccountId, BlockNumber, Balance> {
+        pub user: AccountId,
+        pub block: BlockNumber,
+        pub balance: Balance,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn withdrawal_request_queue)]
+    pub(super) type WithdrawalRequestQueue<T: Config> =
+        StorageValue<_, Vec<Request<T::AccountId, T::BlockNumber, BalanceOf<T>>>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn staked_collators)]
+    pub(super) type StakedCollators<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, BalanceOf<T>, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        /// amount of token to keep in the pot for paying gas fees
+        pub gas_reserve: BalanceOf<T>,
+        pub min_deposit: BalanceOf<T>,
+        pub min_withdraw: BalanceOf<T>,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                min_deposit: 1u32.into(),
+                min_withdraw: 1u32.into(),
+                gas_reserve: 10_000u32.into(),
```

Comment:
we should log a task to not forget to update these once we launch on both Manta and Calamari

Reply:
thanks Georgi, this reminded me that we need to have extrinsics that can actually set these in the pallet lol